psi4/src/psi4/libmints/extern.cc

```diff
@@ -97,28 +97,74 @@ SharedMatrix ExternalPotential::computePotentialMatrix(std::shared_ptr<BasisSet>
     auto V = std::make_shared<Matrix>("External Potential", n, n);
     auto fact = std::make_shared<IntegralFactory>(basis, basis, basis, basis);
 
+    // Thread count
+    int nthreads = 1;
+#ifdef _OPENMP
+    nthreads = Process::environment.get_n_threads();
+#endif
+
     double convfac = 1.0;
     if (basis->molecule()->units() == Molecule::Angstrom) convfac /= pc_bohr2angstroms;
 
     // Monopoles
-    auto V_charge = std::make_shared<Matrix>("External Potential (Charges)", n, n);
-
     auto Zxyz = std::make_shared<Matrix>("Charges (Z,x,y,z)", charges_.size(), 4);
     double **Zxyzp = Zxyz->pointer();
-    for (size_t i = 0; i < charges_.size(); i++) {
+    for (size_t i = 0; i < charges_.size(); ++i) {
         Zxyzp[i][0] = std::get<0>(charges_[i]);
         Zxyzp[i][1] = convfac * std::get<1>(charges_[i]);
         Zxyzp[i][2] = convfac * std::get<2>(charges_[i]);
         Zxyzp[i][3] = convfac * std::get<3>(charges_[i]);
     }
 
-    std::shared_ptr<PotentialInt> pot(static_cast<PotentialInt *>(fact->ao_potential()));
-    pot->set_charge_field(Zxyz);
-    pot->compute(V_charge);
+    std::vector<SharedMatrix> V_charge;
+    std::vector<std::shared_ptr<PotentialInt> > pot;
+    for (size_t t = 0; t < nthreads; ++t) {
+        V_charge.push_back(std::make_shared<Matrix>("External Potential (Charges)", n, n));
+        V_charge[t]->zero();
+        pot.push_back(std::shared_ptr<PotentialInt>(static_cast<PotentialInt *>(fact->ao_potential())));
+        pot[t]->set_charge_field(Zxyz);
+    }
+
+    // Monopole potential is symmetric, so generate unique pairs of shells
+    std::vector<std::pair<size_t, size_t> > ij_pairs;
+    for (size_t i = 0; i < basis->nshell(); ++i) {
+        for (size_t j = 0; j <= i; ++j) {
+            ij_pairs.push_back(std::pair<size_t, size_t>(i, j));
+        }
+    }
+
+    // Calculate monopole potential
+#pragma omp parallel for schedule(guided) num_threads(nthreads)
```

Comment:
The size of each ij-pair is unknown to the scheduler so it's hardly possible to determine a good chunk-size.
Not sure how they are ordered but I'd assume they increase in angular momentum and thus the later loops become more expensive than the earlier.
Probably worth experimenting which scheduler is best.  `dynamic,1` has overhead in creating threads but there would be no waiting on expensive loops.

Reply:
I know you changed the scheduler with the new table, but you also made other changes.